_build/Extension-SetIdentityAndVersion.ps1

```diff
@@ -0,0 +1,198 @@
+# THIS SCRITP WILL NEED TO BE SPLITED NEXT TIME, GETTING TO LONG.
+[CmdletBinding()]
+Param(
+  [Parameter(Mandatory=$True,Position=1)]
+  [string]$vsixManifestFile,
+
+  [Parameter(Mandatory=$True,Position=2)]
+  [string]$vsixIdentity,
+
+  [Parameter(Mandatory=$True,Position=3)]
+  [string]$vsixDisplayName,
+
+  [Parameter(Mandatory=$True,Position=4)]
+  [string]$buildNumber,
+
+  [Parameter(Mandatory=$True,Position=5)]
+  [string]$vsixCommandMenuName,
+
+  [Parameter(Mandatory=$True,Position=6)]
+  [string]$packageGuid,
+
+  [Parameter(Mandatory=$True,Position=7)]
+  [string]$cmdSetGuid,
+
+  [Parameter(Mandatory=$False,Position=8)]
+  [string]$targetVsixCommandMenuName = "Rapid XAML (local)",
+
+  [Parameter(Mandatory=$False,Position=9)]
+  [string]$targetPackageGuid = "c735dfc3-c416-4501-bc33-558e2aaad8c5",
+
+  [Parameter(Mandatory=$False,Position=10)]
+  [string]$targetCmdSetGuid = "8c20aab1-50b0-4523-8d9d-24d512fa8154"
+)
+
+
+$VersionRegex = "(\d+)\.(\d+)\.(\d+)\.(\d+)"
+
+if($buildNumber -match $VersionRegEx){
+
+  Write-Output "Parsed Date From Build: $dateFromBuildNumber"
+
+  $revision =  [int]::Parse($matches[4]).ToString()
+  
+  $versionNumber = [int]::Parse($matches[1]).ToString() + "." + [int]::Parse($matches[2]).ToString() + "." + [int]::Parse($matches[3]).ToString() + "." + $revision
+  Write-Host "Version Number" $versionNumber
+  
+}
+else{
+	throw "Build format does not match the expected pattern (buildName_w.x.y.z)"
+}
+
+## SET IDENTITY AND VERSION IN VSIX Manifest
+if($vsixIdentity){
+  Write-Host
+  Write-Host "Setting Identity in VSIX manifest"
+  if(Test-Path($vsixManifestFile)){
+    [xml]$manifestContent = Get-Content $vsixManifestFile
+    $manifestContent.PackageManifest.Metadata.Identity.Id = $vsixIdentity
+    $manifestContent.PackageManifest.Metadata.Identity.Version = $versionNumber
+    $manifestContent.PackageManifest.Metadata.DisplayName = $vsixDisplayName
+    $resolvedPath = Resolve-Path($vsixManifestFile)
+    $manifestContent.Save($resolvedPath) 
+    Write-Host "$resolvedPath - Version, Identity & DisplayName applied ($versionNumber, $vsixIdentity, $vsixDisplayName)"
+
+    $vsixLangPacks = Get-ChildItem -include "Extension.vsixlangpack" -recurse |  Where-Object{ 
+        $_.FullName -notmatch "\\Templates\\" -and 
+        $_.FullName -notmatch "\\debug\\" -and
+        $_.FullName -notmatch "\\obj\\" -and
+        $_.FullName -match "\\RapidXamlToolkit\\"
+    }
+    if($vsixLangPacks){ 
+      Write-Host "Applying Display Name to vsixlangpack files"
+      foreach ($langPack in $vsixLangPacks) {
+        [xml]$langContent = Get-Content $langPack
+        $langContent.VsixLanguagePack.LocalizedName = $vsixDisplayName
+        $langContent.Save($langPack) 
+        Write-Host "$langPack - LocalizedName applied ($vsixDisplayName)"        
+      }
+    }
+  }
+  else{
+    throw "No VSIX manifest file found."
+  }
+}
+else{
+  throw "Identity is mandatory."
```

Comment:
Wouldn't this be caught when the param is not passed, as it's marked as mandatory?